service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -390,4 +392,40 @@ public String enablePetServiceAccountImpersonation(
       throw new RuntimeException("Error enabling user's pet SA", e);
     }
   }
+
+  /**
+   * Remove a workspace role from a user. This will also remove a user from their private resources
+   * if they are no longer a member of the workspace (i.e. have no other roles) after role removal.
+   *
+   * @param workspaceId ID of the workspace user to remove user's role in
+   * @param role Role to remove
+   * @param userEmail Email identifier of user whose role is being removed
+   * @param userRequest User credentials to authenticate this removal. Must belong to a workspace
```

Comment:
aside: I wonder if there's a convention we could use to make this more prominent. Here we're acting as one user to remove another. Above, we are the service account acting on behalf of (but not "acting as") a user. The documentation is helpful.